### PR TITLE
Fix dagre.js for abnormal wide edges

### PR DESCRIPTION
**Pull request by seanshpark**
This will dagre.js for abnormal wide edges

### CODE REVIEW

**Comment by seanshpark** [comment]
this is to fix #864 issue

File: source/dagre.js [hunk 1]
```diff
@@ -1591,7 +1591,7 @@ dagre.layout = (graph, layout) => {
                             }
                             const label = g.node(v).label;
                             if (min !== Number.POSITIVE_INFINITY && label.borderType !== borderType) {
-                                xs[v] = Math.max(xs[v], min);
+                                xs[v] = Math.min(xs[v], min);
```
Comment (seanshpark) [inline source/dagre.js:1594]: this will choose smallest width

File: source/dagre.js [hunk 2]
```diff
@@ -1740,7 +1740,7 @@ dagre.layout = (graph, layout) => {
                 for (const [v, x] of Object.entries(xs)) {
                     const halfWidth = g.node(v).label.width / 2;
                     max = Math.max(x + halfWidth, max);
-                    min = Math.min(x - halfWidth, min);
+                    min = Math.min(Math.max(0, x - halfWidth), min);
```
Comment (seanshpark) [inline source/dagre.js:1743]: some `x - halfWidth` goes to minus, which will make below `const width = max - min;` larger.

**Comment by lutzroeder** [comment]
This is changing how other models are rendered? [sample.zip](https://github.com/tensorflow/tensorflow/files/917065/sample.zip)

Before:



After:

**Comment by seanshpark** [comment]
> Is there a good way to reason pros vs. cons?

@ lutzroeder, thanks for looking into this !
I'll take a look at the sample model :)

**Comment by seanshpark** [comment]
@lutzroeder , as not knowing much about algorithms in dagre, my wild guess is that the end node of the example graph tries to go to the center of the layout (the minimum diff from center?)

I'll try to study what the algorithm does and find a way for this.
If the example is important, you can just ignore this PR,
or if most of the well know models work, I would be glad if you can land this change.

Anyway, I'll try to find something better.

**Comment by seanshpark** [comment]
Close this for now and reopen when change is ready.

File: source/dagre.js (hunk 1 above)
Comment (beru) [reply source/dagre.js:1594]: @seanshpark @lutzroeder 

How about the following code?

```js
                            let minOut = Number.POSITIVE_INFINITY;
                            let minIn = Number.POSITIVE_INFINITY;
                            for (const e of blockG.node(v).out) {
                                minOut = Math.min(minOut, xs[e.w] - e.label);
                            }
                            for (const e of blockG.node(v).in) {
                                minIn = Math.min(minIn, xs[e.w] - e.label);
                            }
                            const label = g.node(v).label;
                            if (minOut !== Number.POSITIVE_INFINITY && minIn !== Number.NEGATIVE_INFINITY && label.borderType !== borderType) {
                                if (Math.abs(xs[v] - minOut) < Math.abs(xs[v] - minIn)) {
                                    xs[v] = Math.max(xs[v], minOut);
                                }else {
                                    xs[v] = Math.max(xs[v], minIn);
                                }
                            }
```

model | before | after
------|--------|-------
yolov5n | yolov5n onnx | new_yolov5n onnx
sample.zip | saved_model pbtxt | new_saved_model pbtxt

Comment (seanshpark) [reply source/dagre.js:1594]: > How about the following code?

This looks better !

**Comment by seanshpark** [comment]
I couldn't repoen this so posted another  #1254

**Pull request closed without merging**